Source/Core/DolphinQt/FIFO/FIFOAnalyzer.cpp

```diff
@@ -330,42 +348,54 @@ void FIFOAnalyzer::UpdateDetails()
     break;
 
     default:
-      new_label = tr("Unexpected 0x80 call? Aborting...");
-      object_offset = object_nonprim_size;
-      break;
-    }
-    new_label = QStringLiteral("%1:  ").arg(start_offset, 8, 16, QLatin1Char('0')) + new_label;
-    m_detail_list->addItem(new_label);
-  }
+      if ((command & 0xC0) == 0x80)
+      {
+        // Object primitive data
 
-  // Object primitive data
-  ASSERT(object_offset == object_nonprim_size);
-  m_object_data_offsets.push_back(object_offset);
+        const u8 vat = command & OpcodeDecoder::GX_VAT_MASK;
+        const auto& vtx_desc = frame_info.objectCPStates[object_nr].vtxDesc;
+        const auto& vtx_attr = frame_info.objectCPStates[object_nr].vtxAttr[vat];
 
-  const u8 cmd = object[object_offset++];
-  const u16 vertex_count = Common::swap16(&object[object_offset]);
-  object_offset += 2;
+        const auto name = GetPrimitiveName(command);
 
-  const u32 object_prim_size = object_size - object_offset;
+        const u16 vertex_count = Common::swap16(&object[object_offset]);
+        object_offset += 2;
+        const u32 vertex_size = VertexLoaderBase::GetVertexSize(vtx_desc, vtx_attr);
 
-  QString new_label = QStringLiteral("%1:  %2 %3  ")
-                          .arg(object_offset, 8, 16, QLatin1Char('0'))
-                          .arg(cmd, 2, 16, QLatin1Char('0'))
-                          .arg(vertex_count, 4, 16, QLatin1Char('0'));
+        // Note that vertex_count is allowed to be 0, with no special treatment
+        // (another command just comes right after the current command, with no vertices in between)
+        const u32 object_prim_size = vertex_count * vertex_size;
 
-  while (object_offset < object_size)
-  {
-    u32 byte = object[object_offset++];
-    new_label += QStringLiteral("%1").arg(byte, 2, 16, QLatin1Char('0'));
-  }
+        new_label = QStringLiteral("PRIMITIVE %1 (%2)  %3 vertices %4 bytes/vertex %5 total bytes")
+                        .arg(QString::fromStdString(name))
+                        .arg(command, 2, 16, QLatin1Char('0'))
+                        .arg(vertex_count)
+                        .arg(vertex_size)
+                        .arg(object_prim_size);
 
-  if (vertex_count != 0 && (object_prim_size % vertex_count) != 0)
-  {
-    new_label += QLatin1Char{'\n'};
-    new_label += tr("NOTE: Stream size doesn't match actual data length");
+        // It's not really useful to have a massive unreadable hex string for the object primitives.
+        // Put it in the description instead.
+
+        // new_label += QStringLiteral("   ");
```

Comment:
Still a todo or should this be removed?

Reply:
More of a leftover/sample implementation in case it's something that is wanted for the future.  Perhaps `#if 0` would be better than commenting it out?

Reply:
I've replaced it with a `#ifdef` (and also adjusted the implementation, since it wasn't actually correct for multiple objects).